fem/nonlinearform_ext.hpp

```diff
@@ -37,8 +37,22 @@ class PANonlinearFormExtension : public NonlinearFormExtension
    const Operator *elem_restrict_lex; // Not owned
 public:
    PANonlinearFormExtension(NonlinearForm*);
-   void AssemblePA();
+   void Setup();
    void Mult(const Vector &x, Vector &y) const;
 };
+
+/// Data and methods for unassembled nonlinear forms
+class MFNonlinearFormExtension : public NonlinearFormExtension
```

Comment:
Since all nonlinear form representations are "matrix-free", it maybe best to replace `MF` with a different prefix. One option is `UA` (for UnAssembled) -- I've used that abbreviation before, e.g. here: https://ceed.exascaleproject.org/bps/#terminology-and-notation.

What do you think?

Reply:
I think it could be confusing to introduce another acronym, but I don't have a strong opinion.

Reply:
Any other thoughts? @pazner, @barker29, @tzanio?

I see that @YohannDudouit pushed the change `MF` -> `UA` but I'm not sure myself if I like it, so I want to see what everyone thinks.

Reply:
I am probably missing some context here. Would it not be confusing to use `MF` for linear forms and `UA` for nonlinear forms, where they both represent essentially the same "assembly level"?

Reply:
I do not like `UA` (it is not standard and make me think of United Airlines).

I think it is fine to use `MF`. In MFEM we have already said that we are using this for action without storage.